plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -625,15 +627,20 @@ protected ExecutionResult cleanupNetworkElementCommand(final IpAssocCommand cmd)
 
                 // there is only one ip in this public vlan and removing it, so
                 // remove the nic
-                if (ipsCount == 1 && !ip.isAdd()) {
-                    removeVif = true;
+                if (lastIp != null && !ip.isAdd()) {
```

Comment:
Maybe it would make sense to check the value of `lastIp` being something instead of just checking if it's not null? It works the way it is, but perhaps checking specifically for a value like `"true"` would make sense. Would prevent someone from updating code later to send `"false"` for whatever reason and then causing something unexpected. Relying on not null as a truth value is a bit obtuse in my opinion.